padertorch/contrib/examples/tasnet/tasnet.py

```diff
@@ -226,3 +226,54 @@ def review(self, inputs, outputs):
 
     def flatten_parameters(self) -> None:
         self.dprnn.flatten_parameters()
+
+
+class ModularTasNet(TasNet):
+    def __init__(
+            self,
+            encoder,
+            separator,
+            decoder,
+            mask: bool = True,
+            output_nonlinearity: Optional[str] = 'sigmoid',
+            num_speakers: int = 2,
+            additional_out_size: int = 0,
+            sample_rate=8000,
+    ):
+        """
+        Args:
+            encoder:
+            separator:
+            decoder:
+            mask: If `True`, use the output of the NN as a mask in tas domain
+                for separation. Otherwise, use the output directly as an
+                estimation for the separated signals.
+            output_nonlinearity: Nonlinearity applied to the output (right
+                before masking/decoding)
+            num_speakers: The number of speakers/output streams
+            additional_out_size: Size of the additional output. Has no effect if
+                set to 0.
+        """
+        super().__init__()
```

Comment:
Does this work? I think `TasNet.__init__()` will complain.

Reply:
It certainly will

Reply:
this works because tasnet has defaults for all inputs but yes it should be ```super(TasNet).__init__()```

Reply:
I hate `super(TasNet).__init__()`, because it looks like python 2 and it is an indicator to rewrite some code.
As mentioned in another comment changing the inheritance order can solve this.

Reply:
`super(TasNet).__init__()` is wrong (as far as I understand). It should be `super(ModularTasNet, self).__init__()` or better `super().__init__()` and if you really want to skip the `__init__` of `TasNet`, you can write `super(TasNet, self).__init__()`. I generally don't think that skipping a constructor in the MRO is good coding style. And also it looks like Python2.

Reply:
I meant the later one but since we decided to change the inheritance order this problem is already addressed.